CMakeLists.txt

```diff
@@ -239,8 +259,33 @@ if("${FIPS_VERSION}" STREQUAL "v1")
     override_cache(WOLFSSL_TLS13 "no")
 endif()
 
-# TODO: Post-handshake authentication
-#       Hello retry request cookie
+# Post-handshake authentication
+set(WOLFSSL_POSTAUTH_HELP_STRING "Enable wolfSSL Post-handshake Authentication (default: disabled)")
+add_option("WOLFSSL_POSTAUTH" ${WOLFSSL_POSTAUTH_HELP_STRING} "no" "yes;no")
+
+if(WOLFSSL_POSTAUTH)
+    if(NOT WOLFSSL_TLS13)
+        message(WARNING "TLS 1.3 is disabled - disabling Post-handshake Authentication")
+        set(WOLFSSL_POSTAUTH "No") 
```

Comment:
`WOLFSSL_POSTAUTH` is a cache variable, so simply setting it to "No" here won't actually have the desired effect. Check out `override_cache` in functions.cmake.

```
# from wolfSSL source root
$ mkdir build && cd build
$ cmake -DWOLFSSL_TLS13=no -DWOLFSSL_POSTAUTH=yes ..
.
.
.
$ grep "WOLFSSL_POSTAUTH" CMakeCache.txt 
WOLFSSL_POSTAUTH:STRING=yes
//STRINGS property for variable: WOLFSSL_POSTAUTH
WOLFSSL_POSTAUTH-STRINGS:INTERNAL=yes;no
```

Note how `WOLFSSL_POSTAUTH` is still yes. Also, for consistency's sake, drop the capital N and make it "no".

Reply:
Ahh I misssed that. Nice catch.